src/main/java/org/ml_methods_group/ui/AlgorithmsSelectionPanel.java

```diff
@@ -18,53 +18,124 @@
 
 import com.intellij.ui.TitledSeparator;
 import com.intellij.ui.components.JBCheckBox;
+import org.jetbrains.annotations.NotNull;
 import org.ml_methods_group.config.ArchitectureReloadedConfig;
 import org.ml_methods_group.refactoring.RefactoringExecutionContext;
 import org.ml_methods_group.utils.ArchitectureReloadedBundle;
 
 import javax.swing.*;
+import javax.swing.border.Border;
 import java.awt.*;
 import java.util.Set;
 
 public class AlgorithmsSelectionPanel extends JPanel {
-
     public AlgorithmsSelectionPanel() {
         super(new GridBagLayout());
-        final String[] algorithms = RefactoringExecutionContext.getAvailableAlgorithms();
-        final ArchitectureReloadedConfig config = ArchitectureReloadedConfig.getInstance();
+        initializePanel(createRefactoringsTab(), createAlgorithmsTab());
+    }
+
+    private void initializePanel(Tab... tabs) {
+        final GridBagConstraints constraints = new GridBagConstraints();
+        constraints.insets.top = 4;
+        constraints.insets.bottom = 4;
+        constraints.gridx = 0;
+        constraints.gridy = 0;
+        constraints.weightx = 1.0;
+        constraints.weighty = 0.0;
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.anchor = GridBagConstraints.NORTHWEST;
 
         final JComponent separator =
-                new TitledSeparator(ArchitectureReloadedBundle.message("algorithms.selection"));
+            new TitledSeparator(ArchitectureReloadedBundle.message("advanced.settings"));
+
+        add(separator, constraints);
 
+        JTabbedPane tabbedPane = new JTabbedPane();
+
+        Border border = BorderFactory.createEmptyBorder(0, 8, 0, 0);
+        for (Tab tab : tabs) {
+            JPanel wrapper = new JPanel(new FlowLayout(FlowLayout.LEFT));
+            wrapper.add(tab.getComponent());
+            wrapper.setBorder(border);
+            tabbedPane.addTab(tab.getTitle(), wrapper);
+        }
+
+        constraints.gridy++;
+        add(tabbedPane, constraints);
+    }
+
+    private static Tab createRefactoringsTab() {
+        final ArchitectureReloadedConfig config = ArchitectureReloadedConfig.getInstance();
+
+        JPanel panel = new JPanel(new GridBagLayout());
         final GridBagConstraints constraints = new GridBagConstraints();
-        constraints.insets.left = 0;
-        constraints.insets.bottom = 8;
+        constraints.insets.top = 4;
+        constraints.insets.bottom = 4;
+        constraints.gridx = 0;
+        constraints.gridy = 0;
+        constraints.weightx = 1.0;
+        constraints.weighty = 0.0;
+        constraints.fill = GridBagConstraints.HORIZONTAL;
+        constraints.anchor = GridBagConstraints.NORTHWEST;
+
+        final JCheckBox checkBox = new JBCheckBox(
```

Comment:
I think we should explicitly add "Move Method" checkbox here too. Checked by default.

Reply:
As I can see `execute` method of `Algorithm` abstract class has argument `boolean enableFieldRefactorings`. Won't addition of such checkbox require changing interface of `Algorithm` and implementation of all current algorithms?